packages/block-editor/src/store/selectors.js

```diff
@@ -231,6 +210,54 @@ export const getBlocks = createSelector(
 		)
 );
 
+/**
+ * Similar to getBlock, except it will include the entire nested block tree as
+ * inner blocks. The normal getBlock selector will exclude sections of the block
+ * tree which belong to different entities.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Client ID of the block to get.
+ *
+ * @return {Object} The block with all
+ */
+export const __unstableGetBlockWithBlockTree = createSelector(
+	( state, clientId ) => {
+		const block = state.blocks.byClientId[ clientId ];
+		if ( ! block ) {
+			return null;
+		}
+
+		return {
+			...block,
+			attributes: getBlockAttributes( state, clientId ),
+			innerBlocks: __unstableGetBlockTree( state, clientId ),
+		};
+	},
+	( state, clientId ) => [ state.blocks.cache[ clientId ] ]
```

Comment:
I think this is probably wrong since it doesn't account for nested blocks in controlled containers.

```suggestion
	( state ) => [
 		state.blocks.byClientId,
 		state.blocks.order,
 		state.blocks.attributes,
 	]
```

I also wonder if we can merge both `__unstableGetBlockTree` and `__unstableGetBlockWithBlockTree` in a single function.